moveit_core/collision_detection/src/world.cpp

```diff
@@ -174,7 +174,7 @@ const Eigen::Isometry3d& World::getTransform(const std::string& name, bool& fram
   std::map<std::string, ObjectPtr>::const_iterator it = objects_.find(name);
   if (it != objects_.end())
   {
-    if (!it->second->shape_poses_.empty())
+    if (it->second->shape_poses_.size() == 1)
```

Comment:
We should check for `size == 1`, because of this:
https://github.com/ros-planning/moveit/blob/4d3062164bf03ebb74c57d5eba0e3e50438481ba/moveit_core/collision_detection/src/world.cpp#L146

Reply:
Actually back when we last discussed this, we kept the case with multiple `shape_poses_` a gray area on purpose. If you ask for the frame you get the first one, if you ask whether the frame exists, it says no.
I don't have a use-case for objects with multiple shapes, so I don't care too much.

It does seem uncalled for to change this semantic in a segfault fix though. ;)

The proper way to fix this is @felixvd proposal to have a global pose for the object...

Reply:
I cannot believe we agreed on inconsistent behavior on purpose. As both fixes are strongly related, I'm fine to have them in a single PR. IMHO it doesn't make sense to split this tiny PR up into even smaller ones.